shared/templates/shell_lineinfile/tests/quoted.fail.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+FILE_PATH={{{ PATH }}}
+PARAMETER={{{ PARAMETER }}}
+VALUE={{{ VALUE }}}
+NO_QUOTES={{{ NO_QUOTES }}}
+NO_QUOTES=$(echo "$NO_QUOTES" | tr '[:upper:]' '[:lower:]')
+
+if [ "$NO_QUOTES" == "true" ]; then
+    if [ -f "$FILE_PATH" ]; then
+        sed -i "/^\s*${PARAMETER}.*/Id" "$FILE_PATH"
+    else
+        mkdir -p $(dirname "$FILE_PATH")
+        touch "$FILE_PATH"
+    fi
+    echo "${PARAMETER}=\"${VALUE}\"" > "$FILE_PATH"
+else
+{{{ bash_shell_file_set(path=PATH, parameter=PARAMETER, value="badval", no_quotes=NO_QUOTES) }}}
```

Comment:
What if we just inverted `NO_QUOTES`? 
If the rule wants it quoted, we add the value without quotes.
If the rule doesn't want to add quotes, we add the quotes.

Then the test should probably be called inverted_quotes.fail.sh.

Reply:
The problem is that it would fail only on case `no_quotes` is `true` and there are quotes in the `value`, if the `no_quotes` is `false` it doesn't matter if `value` is quoted or not. See https://complianceascode.readthedocs.io/en/latest/templates/template_reference.html?highlight=shell_lineinfile#shell-lineinfile

Reply:
I see, my suggestion doesn't make sense then.